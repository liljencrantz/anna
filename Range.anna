class Range template (T:Int)
{

    var T from;
    var T to;
    var T step;
    
    function __init__(T from, T to, T step)
    {
	this.from = from;
	this.to=to;
	this.step=step;
    }
    
    function __each__(function Null f (Int key, T value))
    {
	var i=this.from;
	var idx = 0;
	
	while(i<this.to)
	{
	    f(idx, i);
	    i = i+this.step;
	    idx++;
	}
	return null;
    }
    
}
