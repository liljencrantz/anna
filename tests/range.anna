
def main(){
    print((0..10).count, "\n"); // 10
    print((0..10|3).count, "\n"); // 4
    print((0..9|3).count, "\n"); // 3
    print((10..0|-3).count, "\n"); // 4
    print((9..0|-3).count, "\n"); // 3
    print((9..0).count, "\n"); // 9
    print((9..0|3).count, "\n"); // ?
    (9..0|3).each(val){print(val)} // Nothing
}
