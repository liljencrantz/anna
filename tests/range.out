10
4
3
4
3
9
null
